pkg/node/snapshot.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2025 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package node
+
+import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"slices"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/core/types"
+	archive "github.com/ethersphere/batch-archive"
+	"github.com/ethersphere/bee/v2/pkg/log"
+)
+
+type SnapshotLogFilterer struct {
+	logger         log.Logger
+	loadedLogs     []types.Log
+	maxBlockHeight uint64
+	isLoaded       bool
+}
+
+func NewSnapshotLogFilterer(logger log.Logger) (*SnapshotLogFilterer, error) {
+	f := &SnapshotLogFilterer{
+		logger: logger,
+	}
+
+	if err := f.loadAndProcessSnapshot(); err != nil {
+		return nil, fmt.Errorf("failed to load and process snapshot during initialization: %w", err)
+	}
+
+	return f, nil
+}
+
+func (f *SnapshotLogFilterer) loadAndProcessSnapshot() error {
+	f.logger.Info("loading batch snapshot during construction")
+	data := archive.GetBatchSnapshot(true)
```

Comment:
archive.GetBatchSnapshot should always provide gzip data to reduce executable binary size.

Reply:
New version tagged https://github.com/ethersphere/batch-archive/releases/tag/v0.0.3 which only provides gzip data